shell/src/main/java/alluxio/master/MasterHealthCheckClient.java

```diff
@@ -166,39 +229,56 @@ public void run() {
   /**
    * Creates a new instance of MasterHealthCheckClient.
    *
-   * @param alluxioMasterName the Alluxio master process name
+   * @param alluxioMasterType the Alluxio master process type
    * @param processCheck whether to check the AlluxioMaster process is alive
+   * @param retryPolicySupplier the policy supplier to utilize for the rpc check
    * @param alluxioConf Alluxio configuration
    */
-  public MasterHealthCheckClient(String alluxioMasterName, boolean processCheck,
-      AlluxioConfiguration alluxioConf) {
-    mAlluxioMasterName = alluxioMasterName;
+  public MasterHealthCheckClient(MasterType alluxioMasterType, boolean processCheck,
+      Supplier<RetryPolicy> retryPolicySupplier, AlluxioConfiguration alluxioConf) {
+    mAlluxioMasterType = alluxioMasterType;
     mProcessCheck = processCheck;
     mExecutorService = Executors.newFixedThreadPool(2);
+    mPolicySupplier = retryPolicySupplier;
     mConf = alluxioConf;
   }
 
   @Override
   public boolean isServing() {
     try {
-      Future<?> masterServingCheckFuture = mExecutorService.submit(
-              new MasterServingCheckRunnable());
+      ServiceType rpcService = ServiceType.FILE_SYSTEM_MASTER_CLIENT_SERVICE;
+      InetSocketAddress connectAddr = NetworkAddressUtils
+          .getConnectAddress(NetworkAddressUtils.ServiceType.MASTER_RPC, mConf);
+      if (MasterType.JOB_MASTER == mAlluxioMasterType) {
+        rpcService = ServiceType.JOB_MASTER_CLIENT_SERVICE;
+        connectAddr = NetworkAddressUtils
+            .getConnectAddress(NetworkAddressUtils.ServiceType.JOB_MASTER_RPC, mConf);
+      }
```

Comment:
Can you turn this into a switch statement with a default case in order to harden it against future master types